docs/proposals/multipart_trasnfer_mode.md

```diff
@@ -0,0 +1,296 @@
+      "oid": "20492a4d0d84f8beb1767f6616229f85d44c2827b64bdbfb260ee12fa1109e0e",
+      "size": 10000000
+    }
+  ]
+}
+```
+
+### Upload Batch Response
+The following is a response for the same request, given an imaginary storage backend:
+
+```json
+{
+  "transfer": "multipart",
+  "objects": [
+    {
+      "oid": "20492a4d0d84f8beb1767f6616229f85d44c2827b64bdbfb260ee12fa1109e0e",
+      "size": 10000000,
+      "actions": {
+        "parts": [
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=0",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 0,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=1",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 2500000,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=2",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 5000000,
+            "size": 2500000,
+            "expires_in": 86400
+          },
+          {
+            "href": "https://foo.cloud.com/storage/upload/20492a4d0d84?part=3",
+            "header": {
+              "Authorization": "Bearer someauthorizationtokenwillbesethere"
+            },
+            "pos": 7500000,
+            "expires_in": 86400
+          }
+        ],
+        "verify": {
+          "href": "https://lfs.mycompany.com/myorg/myrepo/multipart/verify",
+          "authenticated": true,
+          "header": {
+            "Authorization": "Basic 123abc123abc123abc123abc123="
+          },
+          "expires_in": 86400,
+          "params": {
+            "uploadId": "20492a4d0d84",
+            "partIds": [0, 1, 2, 3]
+          }
+        },
+        "abort": {
+          "href": "https://foo.cloud.com/storage/upload/20492a4d0d84",
+          "authenticated": true,
+          "header": {
+            "Authorization": "Basic 123abc123abc123abc123abc123="
+          },
+          "method": "DELETE",
+          "expires_in": 86400
+        }
+      }
+    }
+  ]
+}
+```
+
+## Uploaded Part Digest
+Some storage backends will support, or even require, uploading clients to send a digest of the uploaded part as part
+of the request. This is a useful capability even if not required, as it allows backends to validate each part
+separately as it is uploaded.
+
+To support this, `parts` request objects may include a `want_digest` value, which may be any value specified by
+[RFC-3230](https://tools.ietf.org/html/rfc3230) or [RFC-5843](https://tools.ietf.org/html/rfc5843) (the design for this
+feature is highly inspired by these RFCs).
```

Comment:
Let's specify that it can be any digest registered with IANA via the process outlined in RFC 3230.  Maybe someone wants to specify SHA-3 in a future RFC, and we'd want to allow that.